x-pack/solutions/security/packages/features/src/timeline/kibana_features.ts

```diff
@@ -36,8 +36,8 @@ export const getTimelineBaseKibanaFeature = (
         all: params.savedObjects,
         read: params.savedObjects,
       },
-      ui: ['read', 'crud'],
-      api: ['timeline_read', 'timeline_write'],
+      ui: [],
```

Comment:
shouldn't we make use of ProductFeatureKey.timeline, and ProductFeatureKey.notes to specify these?

Reply:
here: https://github.com/elastic/kibana/blob/7083930b8790d2fbf39b6112198ab64f6b7ca301/x-pack/solutions/security/packages/features/src/security/product_feature_config.ts#L22

Reply:
These are used to check for the timeline feature availability and should not be removed. (search for `securitySolutionTimeline.read` in Kibana)

Reply:
@janmonschke 👍 we'll move these to the product_feature_config.ts so it can make use of PLI as well.

Reply:
It makes more sense, yes. Did you test upgrade scenarios? E.g. someone from 8.18 upgrading to this change in the feature config?

Reply:
In general, what is the approach here for versioning `ProductFeaturesConfig`? In case we wanted to change the timeline / notes config and needed a migration.

Reply:
@janmonschke, from what I understand, roles with deprecated features are always migrated to the last version of each feature. So, I don't think it's necessary to version `ProductFeaturesConfig`. We can ignore deprecated versions, it should always extend the latest version of the feature.

Reply:
> It makes more sense, yes. Did you test upgrade scenarios? E.g. someone from 8.18 upgrading to this change in the feature config?

When this change is deployed to serverless projects users shouldn't notice anything. Currently, only _essentials_ and _complete_ projects exist, which will keep the same capabilities as before.

In classic deployments, all the `ProductFeatureKey` are always enabled except for `ProductFeatureKey.alertsSummary` ([code](https://github.com/elastic/kibana/blob/main/x-pack/solutions/security/plugins/security_solution_ess/common/constants.ts)).
_timeline_ and _notes_ will be always available. So, when upgrading from 8.18, the change will be transparent as well.

Hope it helps

Reply:
Thanks for adding clarity here @semd.

Reply:
@semd , I thought we need to specify timelines and notes in `securityDefaultProductFeaturesConfig`, or isn't it the case here?

Reply:
> So, when upgrading from 8.18, the change will be transparent as well.

It would be great to show that this has been tested or even add an automated test for it.

Reply:
>I thought we need to specify timelines and notes in securityDefaultProductFeaturesConfig, or isn't it the case here?

@tomsonpl Yes and no 😊, they need to be defined in the "productFeaturesConfig", but not in the one of the _Security_ feature. _Timelines_ and _Notes_ have their own features, so they need to be defined in `timelineDefaultProductFeaturesConfig` and `notesDefaultProductFeaturesConfig`, which is already done.

Reply:
Alright, I missed that 👍  thanks!

Reply:
I think I'm more worried about the migration scenario which is hard to test.

Reply:
@janmonschke, we should be good because the Timeline and Notes features are only used by the new roles (`siemV2`+`secTimeline`+`secNotes`), deprecated roles have timeline and notes privileges inside the security feature (`siem`), which has not changed.

But I agree with you that we need to test this PR with a role having the deprecated `siem` feature, and check everything works as expected as well.